### PR TITLE
Added velocity checks and updated to latest php-SDK

### PR DESCRIPTION
**Pull request by ttnnkkrr**
Took the opportunity to update a lot of the code for the sake of readability and to lay the ground work for ease of future enhancements. Velocity monitoring done and tested. Open cart preserves sessions indefinitely and I had to invent a method of keeping these limited to a configurable time without affecting sessions as a whole

### CODE REVIEW

File: catalog/controller/payment/securesubmit.php [hunk 1]
```diff
@@ -40,80 +144,305 @@ public function index() {
...
 
-        $json = array();
+            if ($this->velocityIncrement() < $maxFail){
+                $this->sessionVar('previous',$e->getMessage());
+            }
+            $json['error'] = $e->getMessage();
+        }
+        catch (Exception $e) {
+			$json['error'] = $e->getMessage();
+		}
```
Comment (slogsdon) [inline catalog/controller/payment/securesubmit.php]: @ttnnkkrr Would you mind cleaning up the tabs in this file (I know it wasn't perfect to begin with)? This method in particular is pretty bad. It's a bit hard to read as is.

File: catalog/controller/payment/securesubmit.php [hunk 2]
```diff
@@ -40,80 +144,305 @@ public function index() {
...
+     *
+     * @return mixed
+     */
+    private function sessionVar($key = null, $val = null){
+        /*
+         * I need to set a timebomb on the following variables
+         * $this->session->data[$this->securesubmit_fraud_fail_var]['count']
+         * $this->session->data[$this->securesubmit_fraud_fail_var]['previous']
+         * $this->session->data[$this->securesubmit_fraud_fail_var]['lastErrorTime']
+         */
```
Comment (slogsdon) [inline catalog/controller/payment/securesubmit.php]: @ttnnkkrr Just to confirm, is this session data unique to each user session? It doesn't act like a cache, right?

File: catalog/controller/payment/securesubmit.php (hunk 1 above)
Comment (ttnnkkrr) [inline catalog/controller/payment/securesubmit.php]: sure

File: catalog/controller/payment/securesubmit.php (hunk 2 above)
Comment (ttnnkkrr) [inline catalog/controller/payment/securesubmit.php]: Its IP based and yes tied to the actual session which open cart uses Oddly they favor never expiring _SESSION

**Pull request merged**